tests/Fixer/Phpdoc/PhpdocTypesOrderFixerTest.php

```diff
@@ -0,0 +1,409 @@
+            ],
+            [
+                '<?php /** @var ?\Bar|Foo */',
+                '<?php /** @var Foo|?\Bar */',
+            ],
+            [
+                '<?php /** @var Bar|?\Foo */',
+                '<?php /** @var ?\Foo|Bar */',
+            ],
+        ];
+    }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider getFixWithAlphaAlgorithmAndNullAlwaysLastCases
+     */
+    public function testFixWithAlphaAlgorithmAndNullAlwaysLast($expected, $input = null)
+    {
+        $this->fixer->configure([
+            'sort_algorithm' => 'alpha',
+            'null_position' => 'always_last',
+        ]);
+
+        $this->doTest($expected, $input);
+    }
+
+    public function getFixWithAlphaAlgorithmAndNullAlwaysLastCases()
+    {
+        return [
+            [
+                '<?php /** @var int|string|null */',
+                '<?php /** @var string|int|null */',
+            ],
+            [
+                '<?php /** @param Bar|\Foo */',
+                '<?php /** @param \Foo|Bar */',
+            ],
+            [
+                '<?php /** @property-read \Bar|Foo */',
+                '<?php /** @property-read Foo|\Bar */',
+            ],
+            [
+                '<?php /** @property-write bar|Foo */',
+                '<?php /** @property-write Foo|bar */',
+            ],
+            [
+                '<?php /** @return Bar|foo */',
+                '<?php /** @return foo|Bar */',
+            ],
+            [
+                '<?php /** @method \Bar|Foo foo(\Bar|Foo $foo, \Bar|Foo $bar) */',
+                '<?php /** @method Foo|\Bar foo(Foo|\Bar $foo, Foo|\Bar $bar) */',
+            ],
+            [
+                '<?php /** @var array|Bar\Baz|bool[]|false|Foo|int|object|resource|string|string[]|null */',
+                '<?php /** @var string[]|resource|false|object|null|Foo|Bar\Baz|bool[]|string|array|int */',
+            ],
+            [
+                '<?php /** @var array<int, string>|null Foo */',
+                '<?php /** @var null|array<int, string> Foo */',
+            ],
+            [
+                '<?php /** @var array<int, array<string>>|null Foo */',
+                '<?php /** @var null|array<int, array<string>> Foo */',
+            ],
+            [
+                '<?php /** @var int|string|NULL */',
+                '<?php /** @var string|int|NULL */',
+            ],
+            [
+                '<?php /** @var ?Bar|Foo */',
+                '<?php /** @var Foo|?Bar */',
+            ],
+            [
+                '<?php /** @var Bar|?Foo */',
+                '<?php /** @var ?Foo|Bar */',
+            ],
+            [
+                '<?php /** @var ?\Bar|Foo */',
+                '<?php /** @var Foo|?\Bar */',
+            ],
+            [
+                '<?php /** @var Bar|?\Foo */',
+                '<?php /** @var ?\Foo|Bar */',
+            ],
```

Comment:
please add tests for:
```
@var a null|\null
@var b null|null
@var c \null|null
@var d \null|\null
```

non-should be fixed, non should mark the Tokens collection as being marked as changed.

Please add a test for
```php
@var a \null|int
@var b array<\null|int>
@var c array<array<array<array<\null|int>\null|int>\null|int>\null|int>>>>
```